mne_bids/io.py

```diff
@@ -59,24 +67,40 @@ def _read_raw(raw_fname, electrode=None, hsp=None, hpi=None, config=None,
     elif ext == '.ds':
         raw = io.read_raw_ctf(raw_fname)
 
-    # EEG File Types
+    # EEG and iEEG File Types
     # --------------
     # BrainVision format by Brain Products, links to  a .eeg and a .vmrk file
+    # shared by EEG and iEEG
     elif ext == '.vhdr':
         raw = io.read_raw_brainvision(raw_fname)
 
     # EDF (european data format) or BDF (biosemi) format
+    # shared by EEG and iEEG (.edf) ... only EEG (.bdf)
     elif ext == '.edf' or ext == '.bdf':
         raw = io.read_raw_edf(raw_fname, preload=True)
 
     # EEGLAB .set format, a .fdt file is potentially linked from the .set
+    # shared by EEG and iEEG
     elif ext == '.set':
         raw = io.read_raw_eeglab(raw_fname)
 
     # Neuroscan .cnt format
+    # only EEG
     elif ext == '.cnt':
         raw = io.read_raw_cnt(raw_fname, montage=montage)
 
+    # MEF3 format
+    # only iEEG
+    elif ext == '.mef3':
+        raise ValueError("MEF3 format not yet supported. ",
+                         "Please contact the MNE-BIDS developers.")
+
+    # Neurodata without borders .nwb format
+    # only iEEG
+    elif ext == '.nwb':
+        raise ValueError("Neurodata without borders .nwb format not yet ",
+                         " supported. Please contact the MNE-BIDS developers.")
```

Comment:
isn't this already covered by `ALLOWED_EXTENSIONS` ?

Reply:
ALLOWED_EXTENSIONS raises a different error that might be confusing ... mef3 and nwb are allowed ... they are just not implemented yet